automata/fa/nfa.py

```diff
@@ -238,6 +237,7 @@ def eliminate_lambda(self):
 
     def _check_for_input_rejection(self, current_states):
         """Raise an error if the given config indicates rejected input."""
+        # TODO isn't set.isdisjoint() faster?
         if not (current_states & self.final_states):
```

Comment:
Not sure if it's faster but definitely cleaner + more expressive, let's do it!
```suggestion
        if current_states.isdisjoint(self.final_states):
```

Reply:
@eliotwrobson @khoda81 Just merged #130, so you can remove the TODO here. When that's done, please feel free to mark this comment thread as _Resolved_.

Reply:
It might be faster because when taking `a & b` you are basically doing this: 
```python
{x for x in a if x in b}
```
Which calculates the full intersection to see if there is any intersection. But we can terminate early when we find the first element that is contained in both, so `set.isdisjoint` is kinda like this:
```python
def isdisjoint(a, b):
    for x in a:
        if x in b:
            return False
    return True
```
In theory, it should be faster on large sets, but things might be different in the real-world...